src/codegen/x86_64/arch_x86_64.c

```diff
@@ -1227,10 +1202,12 @@ static void emit_instruction(CodegenContext *context, IRInstruction *inst) {
       // TODO: Should this array to pointer decay happen here? Or higher up in codegen?
       // TODO: type_sizeof(t_pointer) or something to load a pointer sized thing.
       //       WE SHOULD NOT USE t_integer here!!
-      if (inst->operand->type->kind == TYPE_ARRAY) size = regsize_from_bytes(type_sizeof(t_integer));
+      if (inst->operand->type->kind == TYPE_ARRAY || inst->operand->type->pointer.to->kind == TYPE_ARRAY)
+        size = regsize_from_bytes(type_sizeof(t_integer));
```

Comment:
Is this where you wanted to use `type_sizeof(t_pointer)`? :eyes:

Reply:
One of the places, it seems. I only remembered there being one, but I must have copy-pasted haha

Reply:
Implemented in 6615957756a69fd791bd3c3dd316eb4289146fcd